pkg/controller/s3/bucket/taggingConfig.go

```diff
@@ -110,3 +118,36 @@ func (in *TaggingConfigurationClient) Delete(ctx context.Context, bucket *v1beta
 	).Send(ctx)
 	return awsclient.Wrap(err, taggingDeleteFailed)
 }
+
+// LateInitialize does nothing because the resource might have been deleted by
+// the user.
+func (in *TaggingConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketTaggingRequest(&awss3.GetBucketTaggingInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		// Short stop method for requests in a region without Tags
+		if s3.TaggingNotFound(err) {
+			return nil
+		}
+		return awsclient.Wrap(err, taggingGetFailed)
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	// We need the second check here because by default the tags are not set
+	if external.GetBucketTaggingOutput == nil || len(external.TagSet) == 0 {
+		return nil
+	}
+
+	if bucket.Spec.ForProvider.BucketTagging == nil {
+		bucket.Spec.ForProvider.BucketTagging = &v1beta1.Tagging{}
+	}
+
+	bucket.Spec.ForProvider.BucketTagging = GenerateLocalTagging(external.TagSet)
```

Comment:
only if `len(bucket.Spec.ForProvider.BucketTagging.TagSet) == 0 && len(external.TagSet) != 0`

Reply:
Is not sufficient to check `fp.BucketTagging == nil || len(fp.BucketTagging.TagSet) != 0` since we check `len(external.TagSet) == 0` above?